charts/actions-runner-controller/values.yaml

```diff
@@ -38,6 +38,18 @@ securityContext: {}
   # runAsNonRoot: true
   # runAsUser: 1000
 
+# Only 1 authentication method can be enabled at a time
+# Values need to be base64'ed in values.yaml
+githubAuthenticationSchema:
+  app:
+    enabled: false
+    github_app_id: ""
+    github_app_installation_id: ""
+    github_app_private_key: ""
+  token:
+    enabled: false
+    github_token: ""
```

Comment:
```suggestion
# Set `enabled: true` and either specify github_app_* or github_token to let the chart
# create the K8s secret required by the controller to authenticate against GitHub API.
# All the values must NOT be base64-encoded, as the chart template base64-encodes values for you.
authSecret:
  enabled: false
  github_app_id: ""
  github_app_installation_id: ""
  github_app_private_key: ""
  github_token: ""
```

I slightly prefer modeling this around the fact that we just create a single K8s secret containing various credentials.

That would allow us to easily enhance this secret/chart in the future for e.g. automated docker-login to the dockerhub to avoid pull rate limit.

Reply:
In addition to that, `manager_secret.yaml` can be simplified like this:

```
{{- if or .Values.authSecret.enabled }}
apiVersion: v1
kind: Secret
metadata:
  name: controller-manager
  namespace: {{ .Release.Namespace }}
type: Opaque
stringData:
{{ toYaml .Values.authSecret | indent 2 }}
{{- end }} 
```

// I'm aware of the redundant `enabled: true` added to the stringData but had no quick idea to remove that 😃

Reply:
yeh, this makes more sense than the way I was doing it, I'll make the change

Reply:
Patched, resolve if happy